kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceChannelProvider.java

```diff
@@ -0,0 +1,255 @@
+    if (ptvServiceHours == null) {
+      return Collections.emptyList();
+    }
+    
+    List<ServiceChannelServiceHour> result = new ArrayList<>();
+    for (VmOpenApiServiceHour ptvServiceHour : ptvServiceHours) {
+      translateServiceHour(result, ptvServiceHour);
+    }
+    
+    return result;
+  }
+
+  private void translateServiceHour(List<ServiceChannelServiceHour> result, VmOpenApiServiceHour ptvServiceHour) {
+    boolean[] dayBools = { 
+      ptvServiceHour.getMonday(),
+      ptvServiceHour.getTuesday(),
+      ptvServiceHour.getWednesday(),
+      ptvServiceHour.getThursday(),
+      ptvServiceHour.getFriday(),
+      ptvServiceHour.getSaturday(),
+      ptvServiceHour.getSunday()
+    };
+    
+    int dayIndex = 0;
+    List<Integer> days = new ArrayList<>();
+    boolean currentOpen = dayBools[0];
+    
+    while (dayIndex < dayBools.length) {
+      if ((dayBools[dayIndex] != currentOpen) && (!days.isEmpty())) {
+        result.add(createServiceHourObject(ptvServiceHour, days, currentOpen));
+        days.clear();
+        currentOpen = dayBools[dayIndex];
+      } 
+      
+      days.add((dayIndex + 1) % 7);
+      dayIndex++;
+    }
+    
+    if (!days.isEmpty()) {
+      result.add(createServiceHourObject(ptvServiceHour, days, currentOpen));
+    }
+  }
+
+  private ServiceChannelServiceHour createServiceHourObject(VmOpenApiServiceHour ptvServiceHour, List<Integer> days, boolean currentOpen) {
+    ServiceChannelServiceHour serviceHour = new ServiceChannelServiceHour();
+    ArrayList<Integer> openDays = new ArrayList<>(days);
+    
+    serviceHour.setDays(openDays);
+    if (currentOpen) {
+      serviceHour.setOpens(ptvServiceHour.getOpens());
+      serviceHour.setCloses(ptvServiceHour.getCloses());
+    }
+    
+    serviceHour.setStatus(currentOpen ? "OPEN": "CLOSED");
+    serviceHour.setType(ptvServiceHour.getServiceHourType());
+    serviceHour.setValidFrom(ptvServiceHour.getValidFrom());
+    serviceHour.setValidTo(ptvServiceHour.getValidTo());
+    return serviceHour;
+  }
+
+  private List<ServiceChannelSupport> translateSupportContacts(List<VmOpenApiSupport> supportContacts) {
+    if (supportContacts == null) {
+      return Collections.emptyList();
+    }
+    
+    List<ServiceChannelSupport> result = new ArrayList<>(supportContacts.size());
+    
+    for (VmOpenApiSupport supportContact : supportContacts) {
+      ServiceChannelSupport support = new ServiceChannelSupport();
+      support.setEmail(supportContact.getEmail());
+      support.setLanguage(supportContact.getLanguage());
+      support.setPhone(supportContact.getPhone());
+      support.setPhoneChargeDescription(supportContact.getPhoneChargeDescription());
+      support.setServiceChargeTypes(supportContact.getServiceChargeTypes());
+      result.add(support);
+    }
+
+    return result;
+  }
+
+  private List<ServiceChannelAttachment> translateAttachments(List<VmOpenApiAttachmentWithType> ptvAttachments) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateAttachments" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)